### DISCUSSION TITLE
update hydra at kali linux

### DISCUSSION THREAD

**Discussion by shaybuz** [community/community#21738] [New to GitHub]
<p>Hello,</p>
<p>i tried to update the version of hydra at kali linux. how can i perform this step by step?</p>
<p>Best regards,</p>
<p>shaybuz.</p>

**Comment by Gisgar3** [comment, marked as answer]
<p>To update  <strong>Kali Linux</strong> and the applications provided within the operating-system, you can type this in a terminal:</p>
<pre><code>apt-get update &amp;&amp; apt-get upgrade
</code></pre>
<p>^this should get a list of available updates and then upgrade the system to those updates.</p>
<p>Hope this helps!</p>

**Comment by shaybuz** [comment]
<p>Hello Gisgra3,</p>
<p>take a look about this:</p>
<p>root@kali:~# hydra -l <a href="mailto:shiran4489@gmail.com">shiran4489@gmail.com</a> -p /root/Desktop/pass.txt -S -v -V -t 1 <a href="http://imap.gmail.com" rel="nofollow noopener">imap.gmail.com</a> imap<br>
Hydra v8.8 © 2019 by van Hauser/THC - Please do not use in military or secret service organizations, or for illegal purposes.</p>
<p>Hydra (<a href="https://github.com/vanhauser-thc/thc-hydra" rel="nofollow noopener">https://github.com/vanhauser-thc/thc-hydra</a>) starting at 2019-05-14 18:05:10<br>
[WARNING] you want to access SMTP/POP3/IMAP with SSL. Are you sure you want to use direct SSL (-S) instead of STARTTLS (-m TLS)?<br>
[INFO] several providers have implemented cracking protection, check with a small wordlist first - and stay legal!<br>
[WARNING] Google Mail has bruteforce detection and sends false positives. You are not doing anything illegal right?!<br>
[DATA] max 1 task per 1 server, overall 1 task, 1 login try (l:1/p:1), ~1 try per task<br>
[DATA] attacking imaps://imap.gmail.com:993/<br>
[VERBOSE] Resolving addresses … [VERBOSE] resolving done<br>
[ATTEMPT] target <a href="http://imap.gmail.com" rel="nofollow noopener">imap.gmail.com</a> - login “<a href="mailto:shiran4489@gmail.com">shiran4489@gmail.com</a>” - pass “/root/Desktop/pass.txt” - 1 of 1 [child 0] (0/0)<br>
[VERBOSE] CAPABILITY: * CAPABILITY IMAP4rev1 UNSELECT IDLE NAMESPACE QUOTA ID XLIST CHILDREN X-GM-EXT-1 XYZZY SASL-IR AUTH=XOAUTH2 AUTH=PLAIN AUTH=PLAIN-CLIENTTOKEN AUTH=OAUTHBEARER AUTH=XOAUTH<br>
1 OK Thats all she wrote! v2mb106748948wrq<br>
[VERBOSE] using IMAP PLAIN AUTH mechanism<br>
[ERROR] 2 NO [AUTHENTICATIONFAILED] Invalid credentials (Failure)</p>
<p>[STATUS] attack finished for <a href="http://imap.gmail.com" rel="nofollow noopener">imap.gmail.com</a> (waiting for children to complete tests)<br>
1 of 1 target completed, 0 valid passwords found<br>
Hydra (<a href="https://github.com/vanhauser-thc/thc-hydra" rel="nofollow noopener">https://github.com/vanhauser-thc/thc-hydra</a>) finished at 2019-05-14 18:05:12</p>
<p>as u see, the kali direct you to update the version of Hydra throw github. so r u sure about the command u wrote me?</p>

**Comment by shaybuz** [comment]
<p>Hello again,</p>
<p>for your own, i did what u told me and this was the result:</p>
<p>root@kali:~# apt-get update &amp;&amp; apt-get upgrade<br>
Ign:2 <a href="https://apt.syncthing.net" rel="nofollow noopener">https://apt.syncthing.net</a> syncthing InRelease<br>
Ign:3 <a href="https://deb.torproject.org/torproject.org" rel="nofollow noopener">https://deb.torproject.org/torproject.org</a> stretch InRelease<br>
Err:4 <a href="https://apt.syncthing.net" rel="nofollow noopener">https://apt.syncthing.net</a> syncthing Release<br>
Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown. Could not handshake: Error in the certificate verification. [IP: 82.196.13.137 443]<br>
Err:5 <a href="https://deb.torproject.org/torproject.org" rel="nofollow noopener">https://deb.torproject.org/torproject.org</a> stretch Release<br>
Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown. Could not handshake: Error in the certificate verification. [IP: 95.216.163.36 443]<br>
Get:1 <a href="http://kali.download/kali" rel="nofollow noopener">http://kali.download/kali</a> kali-rolling InRelease [30.5 kB]<br>
Get:6 <a href="http://kali.download/kali" rel="nofollow noopener">http://kali.download/kali</a> kali-rolling/main amd64 Packages [17.1 MB]<br>
Reading package lists… Done<br>
E: The repository ‘<a href="https://apt.syncthing.net" rel="nofollow noopener">https://apt.syncthing.net</a> syncthing Release’ does not have a Release file.<br>
N: Updating from such a repository can’t be done securely, and is therefore disabled by default.<br>
N: See apt-secure(8) manpage for repository creation and user configuration details.<br>
E: The repository ‘<a href="https://deb.torproject.org/torproject.org" rel="nofollow noopener">https://deb.torproject.org/torproject.org</a> stretch Release’ does not have a Release file.<br>
N: Updating from such a repository can’t be done securely, and is therefore disabled by default.<br>
N: See apt-secure(8) manpage for repository creation and user configuration details.</p>
<p>kali hard to find a secure repository. i dont know how to find it and set the secure repository as defult. </p>
<p>i will glad for your help about that =]</p>

**Comment by Gisgar3** [comment]
<p>Do you know which version of  <strong>Kali Linux</strong> you’re running? Could be a versioning problem that is causing all of these problems to occur, even ones that prevent you from updating the software itself.</p>

**Comment by shaybuz** [comment]
<p>yes, the version i runing is 6.0.6. it’s the latest one, no?</p>

**Comment by Gisgar3** [comment]
<p>If this version is equal to their <a href="https://www.kali.org/kali-linux-releases/" title="Kali Linux Releases" rel="nofollow noopener">March 4th</a> release, then you’re update to the latest version. I would verify that you have the latest version from their site and see if that fixes anything. Hope it all works!</p>
